### PR TITLE
[MODORG-53] - BE - Implement CRUD for Organization Settings

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://issues.folio.org/browse/MODORG-53

## Approach
Created corresponding table and API

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate action.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code are 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added, or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -762,6 +793,43 @@
...
+      "displayName" : "banking_account_type delete",
+      "description" : "Delete a banking_account_type"
+    },
+    {
+      "permissionName" : "organizations-storage.settings.all",
+      "displayName" : "banking_account_type all",
+      "description" : "All permissions for banking_account_type",
+      "subPermissions" : [
+        "organizations-storage.settings.collection.get",
+        "organizations-storage.settings.item.post",
```
Comment (imerabishvili) [inline descriptors/ModuleDescriptor-template.json:827]: **Question**
Can we have post and delete privileges even API doesn't allow this?

Comment (imerabishvili) [reply descriptors/ModuleDescriptor-template.json]: removed

File: src/main/resources/templates/db_scripts/schema.json [hunk 2]
```diff
@@ -175,6 +180,21 @@
...
+      "tableName": "settings",
+      "fromModuleVersion": "mod-organizations-storage-4.7.0",
+      "withMetadata": true,
+      "fullTextIndex" : [
+        {
+          "fieldName" : "key",
+          "tOps" : "ADD"
+        },
+        {
+          "fieldName" : "value",
```
Comment (SerhiiNosko) [inline src/main/resources/templates/db_scripts/schema.json:194]: I think we don't need index for value field, it will not be used

Comment (SerhiiNosko) [reply src/main/resources/templates/db_scripts/schema.json:194]: also don't sure that we need fullTextIndex here and not just common btree index. 
Basically I don't think that we will have thousands of settings, it is ok to not use indexes here at all

File: src/main/resources/templates/db_scripts/migration/create_banking_information_setting.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO ${myuniversity}_${mymodule}.settings (id, jsonb)
+VALUES ('3c7fc2ac-6d09-11ee-b962-0242ac120002', '{"id": "3c7fc2ac-6d09-11ee-b962-0242ac120002", "key": "BANKING_INFORMATION_ENABLED", "value": "true"}')
```
[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/resources/templates/db_scripts/migration/create_banking_information_setting.sql:2]: and it is sensitive information, by default sensitive indormation should be disabled

[the remark this answers is not in the document]
Comment (imerabishvili) [reply src/main/resources/templates/db_scripts/migration/create_banking_information_setting.sql:2]: Can you link me an example?

[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/resources/templates/db_scripts/migration/create_banking_information_setting.sql:2]: https://www.javainuse.com/uidvalidate

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Review by Dmitriy-Butramyou — APPROVED** [review]

**Review by Abdulkhakimov — APPROVED** [review]

**Pull request merged**